Add automatic relation of SO line and logisitc route based on sourced_by...

nbessi opened the pull request:
... field.

Wee look for the PO related to current SO line by the sourced_by fields.

If the PO has a destination location with usage "customer" we apply the dropshipping route to SO line.

If the PO has a destination location with usage "internal" we apply the make to order route to SO line.

nbessi committed:
Add automatic relation of SO line and logisitc route based on sourced_by field.

Wee look for the PO related to current SO line by the sourced_by fields.

If the PO has a destination location with usage "customer" we apply the dropshipping route to SO line.

If the PO has a destination location with usage "internal" we apply the make to order route to SO line.

jgrandguillaume commented:
@nbessi Can you fix travis please ?

nbessi commented:
@jgrandguillaume not that week. I have asked @lepistone  to have a look

lepistone commented:
Cool, travis is red because of a real functionality change and not becase or some weird incompatibility. It was checking for a warning in case we sourced a sale order line with dropshipping route with an rfq with destination stock.

This PR automatically changes the route for us, so we have no warning anymore.

lepistone committed:
to help me understand a failing test, rephrase it

To figure out what is the assertion, I used some harmless gherkin-like
wording.

lepistone committed:
adapt the test to test the new onchange

We are now green again, and the test looks more readable to me.

On the other hand, we lost the test for the warning in case of
inconsistent PO destination as SOL route.

I am not convinced this check is still necessary. It does no harm still:
we seem to always follow the happy path.

lepistone committed:
document why the dropshipping dep was added again

lepistone commented:
Now the new test is green and the new functionality is covered. I updated the documentation to explain why the stock_dropshipping dependency was added again.

It's now :+1: for me, thanks!

jgrandguillaume commented:
:+1:  Thanks for the work ! I'll merge that.
